processor/src/main/java/org/mapstruct/ap/internal/model/source/MappingMethodOptions.java

```diff
@@ -46,22 +50,25 @@
     private EnumMappingOptions enumMappingOptions;
     private List<ValueMappingOptions> valueMappings;
     private boolean fullyInitialized;
-    private Set<SubclassMappingOptions> subclassMapping;
+    private Map<TypeMirror, SubclassMappingOptions> subclassMapping;
```

Comment:
We can't use `TypeMirror` as a key in a Map. The `equals` implementation of `TypeMirror` might return `false` for types that are actually the same. That's why we use `Types#isSameType` to check these things. Can we not use the FQN of the type instead?

Reply:
Map not needed anymore, implemented an equals on SubclassMappingOptions that uses TypeUtils.isSameType with source MirrorType.